doc: add more details for localStorage and sessionStorage (#53881)
nodejs/node: doc/api/globals.md

The following variables may appear to be global but are not. They exist only in
the scope of [CommonJS modules][]:

- [`__dirname`][]
- [`__filename`][]
- [`exports`][]
- [`module`][]

> - [`require()`][]

Reviewer: When you make Lint, it will switch to the format suitable for ci

```js
const ac = new AbortController();

> ac.signal.addEventListener("abort", () => console.log("Aborted!"), {
>   once: true,
> });

Reviewer 2: ```suggestion
ac.signal.addEventListener('abort', () => console.log('Aborted!'),
                           { once: true });
```

ac.abort();

> console.log(ac.signal.aborted); // Prints true

Reviewer 2: ```suggestion
console.log(ac.signal.aborted);  // Prints true
```

const ac = new AbortController();

// Use either the onabort property...

> ac.signal.onabort = () => console.log("aborted!");

Reviewer 2: ```suggestion
ac.signal.onabort = () => console.log('aborted!');
```

// Or the EventTarget API...

> ac.signal.addEventListener(
>   "abort",
>   (event) => {
>     console.log(event.type); // Prints 'abort'
>   },
>   { once: true }
> );

Reviewer 2: ```suggestion
ac.signal.addEventListener('abort', (event) => {
  console.log(event.type);  // Prints 'abort'
}, { once: true });
```

```js
const ac = new AbortController();

> ac.abort(new Error("boom!"));
> console.log(ac.signal.reason); // Error: boom!

Reviewer 2: ```suggestion
ac.abort(new Error('boom!'));
console.log(ac.signal.reason);  // Error: boom!
```

A browser-compatible implementation of [`localStorage`][]. Data is stored
unencrypted in the file specified by the [`--localstorage-file`][] CLI flag.
Data is stored in memory, with a storage quota of 10 MB.
`localStorage` data persists within the current browser process and across
page refreshes for the same origin,but it's not shared across different

> browser tabs, windows, or server-side rendering (SSR) contexts.

Reviewer 2: What was "current browser" mean? The storage isn't in the browser, but in Node.js, unless I'm missing something

Author: Need to be `current browser's process`. I'll update it.

Reviewer 2: No, I'm not understanding why a browser is even mentioned. I might be missing something, but isn't this done completely independent of the browser?

Author: I get it now. `current browser` is not true on that. It shouldn't be there.

Reviewer 2: While browser is removed, there aren't any 'tabs', 'windows', or 'contexts' in Node.js. (Along with other things mentioned). Could you reword this to fit Node.js instead of the Browser? Thanks ! :-)

> A browser-compatible implementation of [`localStorage`][]. Data is stored

Reviewer 3: I think this needs to mention that both `localStorage` and `sessionStorage` are not per user or per request (when in the context of a server).

unencrypted in the file specified by the [`--localstorage-file`][] CLI flag.

> The maximum amount of data that can be stored is 10MB.

Reviewer 3: For consistency with the `sessionStorage` text.

```suggestion
The maximum amount of data that can be stored is 10 MB.
```

A browser-compatible implementation of [`localStorage`][]. Data is stored

> unencrypted in the file specified by the [`--localstorage-file`][] CLI flag.
> Data is stored in memory, with a storage quota of 10 MB.
> `localStorage` data persists within the current process, but it's not
> shared across different instances or server-side rendering (SSR) contexts.

Reviewer 2: ```suggestion
unencrypted in the file specified by the [`--localstorage-file`][] CLI flag.
The maximum amount of data that can be stored is 10MB.
```

A browser-compatible implementation of [`localStorage`][]. Data is stored
unencrypted in the file specified by the [`--localstorage-file`][] CLI flag.
The maximum amount of data that can be stored is 10 MB.
Any modification of this data outside of the Web Storage API is not supported.
Enable this API with the [`--experimental-webstorage`][] CLI flag.
`localStorage` data is not stored per user or per request when used in the context

> of a server, it is shared across all users and requests.

Reviewer 2: WDYT about this extra bit of information, is this implied?

Author: @cjihrig asked for [this.](https://github.com/nodejs/node/pull/53881#discussion_r1683331170)

logical processors available to the current Node.js instance.

```js

> console.log(
>   `This process is running on ${navigator.hardwareConcurrency} logical processors`
> );

Reviewer 2: ```suggestion
console.log(`This process is running on ${navigator.hardwareConcurrency} logical processors`);
```

The fallback value on builds without ICU is `'en-US'`.

```js

> console.log(
>   `The preferred language of the Node.js instance has the tag '${navigator.language}'`
> );

Reviewer 2: ```suggestion
console.log(`The preferred language of the Node.js instance has the tag '${navigator.language}'`);
```

  const hit = this._cache.get(key);
  if (hit !== undefined) {
    queueMicrotask(() => {

>       this.emit("load", hit);

Reviewer 2: ```suggestion
      this.emit('load', hit);
```

    });
    return;
  }

  const data = await fetchData(key);
  this._cache.set(key, data);

>   this.emit("load", data);

Reviewer 2: ```suggestion
  this.emit('load', data);
```

> Stability: 1.0 - Early development.

A browser-compatible implementation of [`sessionStorage`][]. Data is stored in

> memory, with a storage quota of 10 MB. `sessionStorage` data persists within the
> current process and only for the duration of the current runtime session,
> not across restarts or multiple instances. Any modification of this data
> outside of the Web Storage API is not supported. Enable this API with the

Reviewer 2: ```suggestion
memory, with a storage quota of 10 MB. `sessionStorage` data persists only within
the currently running process, and is not shared between workers.
```

> Stability: 1.0 - Early development.

A browser-compatible implementation of [`sessionStorage`][]. Data is stored in
memory, with a storage quota of 10 MB. `sessionStorage` data persists only within
the currently running process, and is not shared between workers.

> [`--experimental-webstorage`][] CLI flag. `sessionStorage` data is not stored per
> user or per request when used in the context of a server, it is specific to
> a single context.

Reviewer 2: ```suggestion
```

Author: @cjihrig What do you think ?

> Stability: 1.0 - Early development.

A browser-compatible implementation of [`sessionStorage`][]. Data is stored in
memory, with a storage quota of 10 MB. `sessionStorage` data persists only within
the currently running process, and is not shared between workers.
[`--experimental-webstorage`][] CLI flag. `sessionStorage` data is not stored per
user or per request when used in the context of a server, it is specific to

> a single session.

Reviewer 2: While it is called `sessionStorage`, there really isn't a 'session' in Node.js, so saying "it is specific to
a single session." might not make sense.

Author: What can we say about it ?

Reviewer 2: I'm not sure, WDYT @cjihrig

Author: `['--experimental-webstorage'][] CLI flag. When using 'sessionStorage' in the context of a server, it is important to note that the data is not stored per user or per request, but rather it is specific to a single server-side context or instance.`

Maybe this ?

Author: @RedYetiDev @cjihrig

Reviewer 3: ```suggestion
[`--experimental-webstorage`][] CLI flag. `sessionStorage` data is not stored per
user or per request when used in the context of a server, it is specific to
a single context.
```

[buffer section]: buffer.md
[built-in objects]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
[timers]: timers.md

> [webassembly-mdn]: https://developer.mozilla.org/en-US/docs/WebAssembly

Reviewer 2: ```suggestion
[webassembly-mdn]: https://developer.mozilla.org/en-US/docs/WebAssembly

```
One more unrelated change to fix

Author: It is not deleted.

> [webassembly-mdn]: https://developer.mozilla.org/en-US/docs/WebAssembly

Reviewer 3: These changes seem incorrect.

Author: Do we need to keep these two ?

Reviewer 3: They are both still used in the file, so I believe so. In general though, you should avoid touching unrelated code in PRs.

Reviewer 2: My unrelated change fixes didn't resolve that issue, my bad!

[built-in objects]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
[timers]: timers.md
[webassembly-mdn]: https://developer.mozilla.org/en-US/docs/WebAssembly

> [webassembly-org]: https://webassembly.org

Reviewer 2: ```suggestion
```
